File: C/prime.c
#include <stdio.h>
#include <string.h>
int main()
{
// Prime factorisation
int num,arr[10],i,j;
printf("Enter a Number to calculate\n");
scanf("%d",&num);

for(i=0;i<num;i++)
{
if(num%i+2==0)
{
arr[i] = num;
}
}
for(i=0;){
printf("\n");
}
return 0;

File: Haskell/prime.hs
prime :: Int -> Bool
prime 0 = False
prime 1 = False
prime n = checkPrime n (n `div` 2)

checkPrime :: Int -> Int -> Bool
checkPrime _ 1 = True
checkPrime num x
    | (num `mod` x == 0) = False
    | otherwise = checkPrime num (x-1)